fe/fe-core/src/main/java/org/apache/doris/common/PatternMatcher.java

```diff
@@ -20,27 +20,58 @@
 import com.google.common.base.Strings;
 import com.google.common.collect.Sets;
 
+import java.util.Locale;
 import java.util.Set;
 import java.util.regex.Pattern;
 
 // Wrap for Java pattern and matcher
 public class PatternMatcher {
+    public static final PatternMatcher MATCH_ANY = new PatternMatcher(Pattern.compile(".*"));
     private Pattern pattern;
+    private String flatPattern;
```

Comment:
Add some comment in code to explain the different between `pattern` and `flatPattern`. Better to give some example.
And looks like the `flatPattern` is not a real pattern, more like an `origin string`?

Reply:
done